java-snapshot-testing-core/src/main/java/au/com/origin/snapshots/SnapshotConfig.java

```diff
@@ -95,4 +96,44 @@ default SnapshotSerializer getSerializer() {
         return new ToStringSnapshotSerializer();
     }
 
+    /**
+     * Optional
+     * Allows you to perform any custom modifications to the file before it is saved
+     *
+     * @param testClass target test class
+     * @param snapshotContent snapshot file as a string
+     * @return snapshot file contents to be persisted
+     */
+    default String onSaveSnapshotFile(Class<?> testClass, String snapshotContent) {
+        return snapshotContent;
+    }
+
+
+    /**
+     * Optional
+     * Override to supply your own custom comparator function
+     *
+     * @return custom comparator function
+     */
+    default SnapshotComparator getComparator() {
+        return new PlainTextEqualsComparator();
+    }
+
+    /**
+     * Optional
+     * Override to supply your own custom diff reporter functions
+     * Reporters will run in the same sequence as provided.
+     * Reporters can choose to throw exceptions in which case subsequent reporters will be skipped.
+     * Typically one such reporter can be included at the end which can use common assertion libraries
+     * like assertj or junit assertions to 'fail' the test. This is useful since IDEs like intellij can
+     * then show a diff using their native diff tools.
+     * <p>
+     * In the absence of any 'throwing' reporters in the list, a default failure exception will be thrown
+     * to make sure tests fail.
+     *
+     * @return custom reporter functions
+     */
+    default List<SnapshotDiffReporter> getDiffReporters() {
```

Comment:
Can this be changed to `getReporters()` and anywhere else `diffReporters` is used updated also.  I'm thinking that some reporters might not necessarily be diffs.

Reply:
Makes sense. Done.